tests/Functional/ContentProjectRenderTest.php

```diff
@@ -228,14 +236,37 @@ public function testProjectRendering(): void {
     $this->assertCount(1, $stakeholder_sub_headers);
     $this->assertEquals($stakeholder_sub_headers[0]->getText(), 'Participants');
     $this->assertStakeholderOrganisationRendering($project_stakeholders, 'participant');
+
+    // Assert Project's contacts.
+    $project_contacts = $this->assertSession()->elementExists('css', 'div#project-contacts');
+    $contact_headers = $project_contacts->findAll('css', 'h2');
+    $this->assertEquals($contact_headers[0]->getText(), 'Contact');
+    $this->assertContactRendering($project_contacts, 'general_contact');
+
+    // Unpublish Contact entity to test its visibility.
+    $general_contact->set('status', CorporateEntityInterface::NOT_PUBLISHED);
+    $general_contact->save();
+
+    // Reload the page.
+    $this->drupalGet($node->toUrl());
+
+    // Asset Contact entity visibility.
+    try {
```

Comment:
Instead of try/catch-ing, which I think should be used when the outcome of the result is not in your control i.e. when writing files or web service apis, we can use:
```suggestion
  $project_contacts = $this->getSession()->getPage()->findAll('css', 'div#project-contacts');
  $this->assertCount(0, $project_contacts);
```

Reply:
As I see it is common practice to use try/catch in the tests. For example - build/core/tests/Drupal/FunctionalTests/BrowserTestBaseTest.php.

Reply:
there they are testing the api, that a assertion fails when it should, I think this is different but if you think its best, then we can do it like that, up to you.